ethers-ffi/src/lib.rs

```diff
@@ -448,18 +410,10 @@ pub mod android {
         // Convert the Rust String to a JString
         let output = env.new_string(signature).expect("Couldn't create java string!");
 
+        // Free signature pointer in C 
+        string_free(signature_ptr);
+
         // Return the JString
         output.into_inner()
     }
-
-    // Function to free a string
-    #[no_mangle]
-    pub extern "system" fn Java_com_uniswap_EthersRs_stringFree(env: JNIEnv, _class: JClass, string: JString) {
-
-        let c_string_str = jstring_to_cstring(&env, string);
-        let c_string_ptr = c_string_str.into_raw(); // makes mutable raw pointer
-
-        // Free the string
-        string_free(c_string_ptr);
-    }
```

Comment:
removed these since we no longer need to free from java if we handle it here in rust.